fe1-web/src/features/popcha/network/PopchaMessageApi.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Contains all functions to send popcha related messages.
+ * */
+
+import { publish } from 'core/network';
+import { Base64UrlData, getPopchaAuthenticationChannel, Hash, PopToken } from 'core/objects';
+
+import { PopchaAuthMsg } from './messages/PopchaAuthMsg';
+
+/**
+ * Sends a message to the server to authenticate that the user belongs to the lao
+ * @param client_id id of the client
+ * @param nonce nonce
+ * @param popcha_address address of the popcha server
+ * @param state state
+ * @param response_mode response mode (optional) ('query' or 'fragment')
+ * @param laoId id of the lao
+ * @param generateToken function to deterministically generate a long term token
+ * @returns A promise that resolves when the message has been sent
+ */
+export const sendPopchaAuthRequest = (
+  client_id: Hash,
+  nonce: string,
+  popcha_address: string,
+  state: string | null,
+  response_mode: string | null,
+  laoId: Hash,
+  // TODO: hook currently throwing error
+  generateToken: (laoId: Hash, clientId: Hash | undefined) => Promise<PopToken>,
```

Comment:
There is currently a problem with the hook generateToken: I can't use in this function, due the error [Hooks can only be called inside the body of a function component](https://react.dev/warnings/invalid-hook-call-warning)
I am not sure of a nice way to fix this

Reply:
The best way is to compute the promise in the component and to pass it to the sendPopchaAuthRequest as an arg. But I am not sure if it will work, since you do not have the client_id before scanning the QR Code

Reply:
Yes that is the problem I have, even if I generate before calling the `sendPopchaAuthRequest`, I still get the same issue. I have some ideas to fix this but I am not too sure it works / it is good practice, I will ask for this on Wednesday

Reply:
oh I see that I haven't removed the comment, but now it works fine. It was an issue on how I defined the hook in the popcha hooks, but now that it is fixed I will remove the comment